src/main/java/httpserver/HttpRequest.java

```diff
@@ -0,0 +1,532 @@
+    }
+
+    /**
+     * Figure out what kind of HttpHandler you want, based on the path. <p>
+     *
+     * This uses the statically set {@link HttpRouter} to determine the
+     * correct HttpHandler to be used for the current request. If there isn't
+     * a statically set HttpRouter, a 500 error is sent back to the
+     * client.
+     *
+     * @return a new instance of some form of HttpHandler.
+     *
+     * @see HttpRouter
+     * @see HttpRouter#determineHandler
+     * @see HttpHandler
+     */
+    public HttpHandler determineHandler() {
+        if (router == null) {
+            return new DeathHandler();
+        }
+
+        String path = getSplitPath().isEmpty() ? "" : getSplitPath().get(0);
+        return router.route(path, this);
+    }
+
+    /**
+     * Return if the request type is the passed in type.
+     * @param requestTypeCheck The type to check.
+     * @return whether the request type equals the passed in String.
+     */
+    public boolean isType(String requestTypeCheck) {
+        return getRequestType().equalsIgnoreCase(requestTypeCheck);
+    }
+
+    /**
+     * Sets the requestLine, and all derived items. <p>
+     *
+     * Based off of the passed in line, the request type, request path, and
+     * request protocol can be set.
+     *
+     * @param line  The first line in an HTTP request. Should be in
+     *              {@code [type] [full path] [protocol]} form.
+     * @throws HttpException  When the first line does not contain two spaces,
+     *                        signifying that the passed in line is not in
+     *                        HTTP 1.1. When the type is not an expected type
+     *                        (currently GET, POST, and HEAD).
+     *
+     * @see HttpRequest#setRequestType
+     * @see HttpRequest#setFullPath
+     * @see HttpRequest#setRequestProtocol
+     */
+    public void setRequestLine(String line) throws HttpException {
+        this.requestLine = line;
+
+        /*  Split apart the request line by spaces, as per the protocol.
+            The request line should be:
+            [request type] [path] [protocol]
+            */
+        String[] splitty = requestLine.trim().split(" ");
+        if (splitty.length != 3) {
+            throw new HttpException("Request line has a number of spaces other than 3.");
+        }
+
+
+        // Set the request type
+        if (splitty[0].equalsIgnoreCase(GET_REQUEST_TYPE)) {
+            setRequestType(GET_REQUEST_TYPE);
+        }
+        else if (splitty[0].equalsIgnoreCase(POST_REQUEST_TYPE)) {
+            setRequestType(POST_REQUEST_TYPE);
+        }
+        else if (splitty[0].equalsIgnoreCase(HEAD_REQUEST_TYPE)) {
+            setRequestType(HEAD_REQUEST_TYPE);
+        }
+        else if (splitty[0].equalsIgnoreCase(DELETE_REQUEST_TYPE)) {
+            setRequestType(DELETE_REQUEST_TYPE);
+        }
+        else {
+            throw new HttpException("Unexpected request type: " + splitty[0]);
```

Comment:
1. Confirm that `GET`, `POST`, `HEAD`, and `DELETE` are the only "real" supported verbs.
2. Disregard that because it doesn't matter, and `HttpHandler` now supports _any_ request verb, official or not.